src/app/components/progress-indicator/waterfall-progress-indicator/index.html

```diff
@@ -0,0 +1,325 @@
+<sky-tabbed-demo-page
+  pageTitle="Waterfall Progress Indicator"
+>
+
+  <sky-demo-page-summary>
+    <p>
+      The progress indicator module visually represents progress through sequential steps toward a goal. Waterfall progress indicators walk users through discrete steps on a page.
+    </p>
+  </sky-demo-page-summary>
+
+  <div class="sky-margin-stacked-content-section">
+    <sky-demo-page-example>
+      <sky-progress-indicator-demo>
+      </sky-progress-indicator-demo>
+    </sky-demo-page-example>
+  </div>
+
+  <sky-tabset id="content-tabs">
+    
+    <sky-tab
+      tabHeading="Design">
+    
+      <sky-demo-page-content sectionHeading="Usage">
+        <h3 class="sky-text-success">Use when</h3>
+        <sky-docs-visual-example>
+          <sky-docs-visual-example-text>
+            <p>Use waterfall progress indicators for long or complex setup processes that users typically complete only once. Waterfall indicators are ideal when events outside of the system force users to complete tasks in multiple sittings or when users must enter large amounts of data.</p>
+            <p>Use waterfall progress indicators when processes include at least three steps and the number and order of steps do not change.</p>
+          </sky-docs-visual-example-text>
+          <sky-docs-visual-example-image>
+            <div>
+              <img src="~/assets/img/guidelines/progress-indicator/waterfall-progress-usage-1.png" />
+              <div class="caption caption-success sky-text-success">Do use waterfall progress indicators for complex configuration processes that users may complete in stages.</div>
+            </div>
+          </sky-docs-visual-example-image>
+        </sky-docs-visual-example>
+        
+        <sky-docs-visual-example>
+          <sky-docs-visual-example-text>
+            <p>Use waterfall progress indicators when the complexity of steps varies drastically. For example, one step may require a full-screen modal for complex interactions, while another step requires a small modal for a few simple inputs. Waterfall steps can open <a routerLink="/components/wizard">wizards</a> if necessary.</p>
```

Comment:
I went to the modal docs to see if “full-screen modal” was the correct terminology. In those docs, we use “full-screen” and “full-page” interchangeably. I think we should we stick to one or the other. Do you know which is preferred?

Reply:
I think "full-screen" is more precise.

Reply:
I left "full-screen" as is here, and I just pushed updates to this branch for the modal docs to use it consistently there too.